lib/tar.ml

```diff
@@ -412,41 +400,36 @@ module Header = struct
          - the <keyword> cannot contain an equals sign
          - the <length> is the number of octets of the record, including \n
         *)
-      let find buffer char =
-        let rec loop i =
-          if i = Cstruct.length buffer
-          then None
-          else if Cstruct.get_char buffer i = char
-          then Some i
-          else loop (i + 1)
-        in
-        loop 0
+      let find start char =
+        try Some (String.index_from c start char) with Not_found -> None
```

Comment:
We could use a result given that we always return the same error and then use `let*` at call sites.
```suggestion
      let find start char =
        try Ok (String.index_from c start char) with Not_found -> Error (`Unmarshal "Failed to decode pax extended header record")
```
but that can be done in a separate PR.